hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/cmd/ExecuteOperationCommand.java

```diff
@@ -19,68 +19,61 @@
 import java.util.Collection;
 import java.util.Map;
 
-import org.hawkular.agent.monitor.extension.MonitorServiceConfiguration;
 import org.hawkular.agent.monitor.inventory.ID;
-import org.hawkular.agent.monitor.inventory.InventoryIdUtil;
-import org.hawkular.agent.monitor.inventory.InventoryIdUtil.ResourceIdParts;
 import org.hawkular.agent.monitor.inventory.ManagedServer;
-import org.hawkular.agent.monitor.inventory.Name;
 import org.hawkular.agent.monitor.inventory.ResourceManager;
 import org.hawkular.agent.monitor.inventory.dmr.DMRInventoryManager;
 import org.hawkular.agent.monitor.inventory.dmr.DMROperation;
 import org.hawkular.agent.monitor.inventory.dmr.DMRResource;
-import org.hawkular.agent.monitor.inventory.dmr.LocalDMRManagedServer;
-import org.hawkular.agent.monitor.inventory.dmr.RemoteDMRManagedServer;
 import org.hawkular.agent.monitor.log.AgentLoggers;
 import org.hawkular.agent.monitor.log.MsgLogger;
 import org.hawkular.bus.common.BasicMessageWithExtraData;
 import org.hawkular.cmdgw.api.ExecuteOperationRequest;
 import org.hawkular.cmdgw.api.ExecuteOperationResponse;
-import org.hawkular.cmdgw.api.MessageUtils;
-import org.hawkular.cmdgw.api.ResponseStatus;
+import org.hawkular.dmr.api.OperationBuilder;
 import org.hawkular.dmrclient.Address;
-import org.hawkular.dmrclient.CoreJBossASClient;
-import org.hawkular.dmrclient.JBossASClient;
 import org.hawkular.inventory.api.model.CanonicalPath;
 import org.jboss.as.controller.client.ModelControllerClient;
-import org.jboss.dmr.ModelNode;
 
 /**
  * Execute an operation on a resource.
  */
-public class ExecuteOperationCommand implements Command<ExecuteOperationRequest, ExecuteOperationResponse> {
+public class ExecuteOperationCommand extends
+        AbstractResourcePathCommand<ExecuteOperationRequest, ExecuteOperationResponse> {
     private static final MsgLogger log = AgentLoggers.getLogger(ExecuteOperationCommand.class);
     public static final Class<ExecuteOperationRequest> REQUEST_CLASS = ExecuteOperationRequest.class;
 
+    public ExecuteOperationCommand() {
+        super("Execute Operation", "DMR Node");
+    }
+
+    /** @see org.hawkular.agent.monitor.cmd.AbstractResourcePathCommand#createResponse() */
     @Override
-    public BasicMessageWithExtraData<ExecuteOperationResponse> execute(
-            BasicMessageWithExtraData<ExecuteOperationRequest> envelope, CommandContext context) throws Exception {
-        ExecuteOperationRequest request = envelope.getBasicMessage();
-        log.infof("Received request to execute operation [%s] on resource [%s]", request.getOperationName(),
```

Comment:
I don't care where the logging happens, I just wanted to make sure that "somewhere" we log the fact that a request came in to execute operation named O on resource R. However best you make that happen (or if it already happens, then great!) is up to you.